src/index.ts

```diff
@@ -378,12 +388,47 @@ function getNestedValue<ItemType>(
   obj: ItemType,
 ): string | Array<string> | null {
   // @ts-expect-error really have no idea how to type this properly...
-  return key.split('.').reduce((itemObj: object | null, nestedKey: string):
+  return key.split('.').reduce((value: object | null, nestedKey: string):
     | object
     | string
     | null => {
-    // @ts-expect-error lost on this one as well...
-    return itemObj ? itemObj[nestedKey] : null
+    if (value == null) {
+      return null
+    }
+
+    if (Object.hasOwnProperty.call(value,nestedKey)) {
+      // @ts-expect-error lost on this one as well...
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const nestedValue = value[nestedKey]
+      if (nestedValue != null) {
+        return nestedValue
+      }
+      return null
+    }
+
+    if (Array.isArray(value)) {
+      if (nestedKey === "*") {
+        // ignore explicit wildcards
+        return value
+      }
+
+      return value.reduce((values: Array<object | string>, arrayValue: object | null):
```

Comment:
This looks a lot like the first part of the function. I wonder if we could use recursion instead of duplication here. Could you try it out and see whether it's any simpler?

Reply:
Sure, see the last commit which is my attempt at doing so. It seems possible without any performance implications. 

However, there is the downside that the implicit wildcard no longer works, except if the array is a top-level property of the object. So `{favoriteIceCream: ['mint', 'chocolate']}` is fine when matching `'favoriteIceCream'`, but `{favorite: {iceCream: ['mint', 'chocolate']}}` won't work with `'favorite.iceCream'` and instead needs `'favorite.iceCream.*'`.  I personally don't care about implicit wildcards: I never added this intentionally. It just worked due to the checks that made this new feature backwards compatible with all tests. And it's these checks that caused the duplication and now are now gone. Supporting nested array values without explicit wildcards again would require quite a bit of duplication I'm afraid.

The README doesn't explicitly say that nested array values are supported. It says that array values are supported and nested values are supported, but nothing about combining that. But as a user I'd assume that they would be, so now this new feature would be a breaking change.